### ISSUE TITLE
sample at Using XA Transactions is missing an include file

### ISSUE THREAD

**Issue by starquestv** [MicrosoftDocs/feedback#2191]
https://docs.microsoft.com/en-us/sql/connect/odbc/use-xa-with-dtc?view=sql-server-ver15
Using XA Transactions

the sample is incomplete - the include file XaTestRunner.h is not supplied.

**Comment by starquestv** [comment]
FYI, it appears that more than just the include file XaTestRunner.h is missing. The XATestRunner class is derived from the TestRunner class, which is also not present.

**Comment by MightyPen** [comment]
@David-Engel Hi David, If you can add a Comment in here in Issue #2191, that would be very helpful.
Thanks, G.

- `https://github.com/MicrosoftDocs/feedback/issues/2191`

cc: @starquestv @MightyPen

**Comment by v-makouz** [comment]
I've added the missing sample code, it's in PR now
https://github.com/MicrosoftDocs/sql-docs-pr/pull/13215

**Comment by starquestv** [comment]
Checking in - wondering how long it takes for something to get from PR (peer review?) to publication?

Meanwhile, our developer made his best guess about what the missing code should do, but I expect this will be an issue for someone else down the road.

**Comment by MightyPen** [comment]
@starquestv Hello,  On 2020/Jan/31, our internal Pull Request  13215 fixed this Issue.
Sorry our process slipped up by not giving you that final piece of info.
Here is one of the three files that PR 13215 edited:
- `https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver15`

Thank you for submitting this Issue. We hope to hear from you again.

cc: @v-makouz

**Comment by MightyPen** [comment]
#please-close

**Comment by v-makouz** [comment]
@MightyPen @starquestv It seems that XA DTC doc changes got lost from that PR that was merged (#13215), I think I resolved a conflict wrong, that simply reverted the xa_dtc file

Here's the new PR: https://github.com/MicrosoftDocs/sql-docs-pr/pull/13874

**Comment by starquestv** [comment]
So how long should I expect to wait before the new PR takes effect? Judging from the previous cycle (time from PR to publication), that looks to be about 6 weeks?

**Comment by v-makouz** [comment]
It should be a lot quicker then that, I've seen merged changes show up the same day or the next day.

**Comment by starquestv** [comment]
checking in - 7 days, still not there.  I'll keep checking.

**Comment by v-makouz** [comment]
@starquestv Hmm, I just looked at it and it shows up for me, I can see XaTestRunner.h:
https://docs.microsoft.com/en-us/sql/connect/odbc/use-xa-with-dtc?view=sql-server-ver15#xatestrunnerh

**Comment by MightyPen** [comment]
I too see XaTestRunner.h .

cc: @v-makouz @starquestv

**Comment by starquestv** [comment]
Right!  I see it too now.   I was going by the date listed at the top of the page, which still shows 02/04/2019.

So good progress.  

I think there is still this missing:
" The XATestRunner class is derived from the TestRunner class, which is also not present."
(i.e. should there also be a TestRunner.cpp?)

**Comment by v-makouz** [comment]
The cpp is there:
https://docs.microsoft.com/en-us/sql/connect/odbc/use-xa-with-dtc?view=sql-server-ver15#xatestrunnercpp

Yeah, I forgot to change the date, I'll try to be more attentive next time, thanks for pointing that out

**Comment by starquestv** [comment]
Brilliant!  Seems like it's all there now (though I haven't gotten any feedback from our developer who discovered the missing code in the first place - he's working on another project now, so this is not his highest priority) - I think we can close this ticket, unless you want to wait a few days for feedback from our developer.

**Comment by WilliamAntonRohm** [comment]
#please-close

**Comment by WilliamAntonRohm** [comment]
@docs-product -- please close this issue.
